src/main/java/co/cask/PluginJSON.java

```diff
@@ -196,6 +204,73 @@ public void execute()  throws MojoExecutionException {
   }
 
   /**
+   * Inspects all the Icon files to add properties to related widget.
+   */
+  private void addIcon(JSONObject finalObject, Map<String, File> files, File nameFile) throws IOException {
+    String name = FileUtils.removeExtension(nameFile.getName());
+    if (files.containsKey(name)) {
+      JSONObject innerObject = new JSONObject();
+      innerObject.put("type", "inline");
+      Map<String, String> arguments = new TreeMap<String, String>();
+      File file = files.get(name);
+      String mediaType = URLConnection.guessContentTypeFromName(file.getName());
+      arguments.put("data", getDataURISchemaAsString(file, mediaType));
+      innerObject.put("arguments", arguments);
+      finalObject.put("icon", innerObject);
+    }
+  }
+
+  /**
+   * Returns the Data URI Scheme of File
+   */
+  private String getDataURISchemaAsString(File file, String mediaType) throws IOException {
+    FileInputStream inStream = null;
+    ByteArrayOutputStream outStream = null;
+    Base64OutputStream outStreamBase64 = null;
+
+    String encodedData;
+    try {
+      inStream = new FileInputStream(file);
+      try {
+        outStream = new ByteArrayOutputStream();
+        outStreamBase64 = new Base64OutputStream(outStream);
+
+        IOUtils.copy(inStream, outStreamBase64);
+        outStreamBase64.flush();
+
+        encodedData = outStream.toString();
+      } finally {
+        outStreamBase64.close();
+      }
+    } finally {
+      inStream.close();
+    }
+
+    return "data:" + mediaType + ";base64," + encodedData;
+  }
+
+  /**
+   * Remove extensions from files in directory and return a map of names to files.
+   */
+  private Map<String, File> getFileNameMap(File[] listFiles) {
+    Map<String, File> names = new TreeMap<String, File>();
```

Comment:
also don't need to declare the generic on the right hand side, `new HashMap<>();` Your IDE should be marking this as a warning, try to keep your code free of warnings.